### PR TITLE
Migrate to tfplugindocs structure

### PR DESCRIPTION
**Pull request by sheneska**
### Description

This PR covers the following:

- Migrating to tfplugin docs structure
- Customizing `helm_release` resource and `template` data source files to include the appropriate `data fields`
- Addresses #1371. 

### Acceptance tests
- [ ] Have you added an acceptance test for the functionality being added?

### Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-helm/blob/main/CHANGELOG.md):

```release-note
...
```
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -141,6 +139,23 @@ provider "helm" {
 }
 ```
 
+For example, to [authenticate with GKE](https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/using_gke_with_terraform#using-the-kubernetes-and-helm-providers), the `gke-cloud-auth-plugin` can be used:
```
Comment (BBBmau) [inline docs/index.md:142]: I think we'd want to move this to a separate PR and also include azure since it's the last provider to be included. Thoughts?

File: scripts/markdown-link-check.sh [hunk 2]
```diff
@@ -35,7 +35,7 @@ ${DOCKER} run ${DOCKER_RUN_OPTS} --rm -t \
   -w /github/workspace \
   --entrypoint /usr/bin/find \
   docker.io/robertbeal/markdown-link-checker \
-  website \( -type f -name "*.md" -or -name "*.markdown" \) -exec markdown-link-check --config .markdownlinkcheck.json --quiet --verbose {} \; \
+  docs \( -type f -name "*.md" -or -name "*.markdown" \) -exec markdown-link-check --config .markdownlinkcheck.json --quiet --verbose {} \; \
```
Comment (BBBmau) [inline scripts/markdown-link-check.sh]: we'll want to update the GHA since it's failing because of the missing files in `docs/d` and `docs/r`

File: docs/index.md (hunk 1 above)
Comment (sheneska) [reply docs/index.md:142]: I agree with including Azure - i'll do that in a separate PR

File: scripts/markdown-link-check.sh [hunk 3]
```diff
@@ -35,7 +35,7 @@ ${DOCKER} run ${DOCKER_RUN_OPTS} --rm -t \
   -w /github/workspace \
   --entrypoint /usr/bin/find \
   docker.io/robertbeal/markdown-link-checker \
-  website \( -type f -name "*.md" -or -name "*.markdown" \) -exec markdown-link-check --config .markdownlinkcheck.json --quiet --verbose {} \; \
+  find docs/resources docs/data-sources \( -type f -name "*.md" -or -name "*.markdown" \) -exec markdown-link-check --config .markdownlinkcheck.json --quiet --verbose {} \; \
```
Comment (bflad) [inline scripts/markdown-link-check.sh]: This should resolve the script error -- when Docker's `--entrypoint` flag is given, any of the arguments passed at the end go directly to the `--entrypoint` command. So in this case the `find` here is redundant and why the output contains:

```
find: find: No such file or directory
```

Which ultimately shows up to the `markdown-link-check.sh` script as an error. 👍 

```suggestion
  docs/resources docs/data-sources \( -type f -name "*.md" -or -name "*.markdown" \) -exec markdown-link-check --config .markdownlinkcheck.json --quiet --verbose {} \; \
```

Running this script locally with this change, I get:

```
./scripts/markdown-link-check.sh docker "" rw $(pwd)
==> Checking Markdown links...
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested

FILE: docs/resources/release.md

FILE: docs/data-sources/template.md
==================> MARKDOWN LINK CHECK SUCCESS <==================

[✔] All links are good!

===================================================================
```

**Review by BBBmau — APPROVED** [review]
nice!

File: docs/resources/release.md [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+- `reuse_values` (Boolean) When upgrading, reuse the last release's values and merge in any overrides. If 'reset_values' is specified, this is ignored
+- `set` (Block Set) Custom values to be merged with the values. (see [below for nested schema](#nestedblock--set))
+- `set_list` (Block List) Custom sensitive values to be merged with the values. (see [below for nested schema](#nestedblock--set_list))
+- `set_sensitive` (Block Set) Custom sensitive values to be merged with the values. (see [below for nested schema](#nestedblock--set_sensitive))
+- `skip_crds` (Boolean) If set, no CRDs will be installed. By default, CRDs are installed if not already present
+- `timeout` (Number) Time in seconds to wait for any individual kubernetes operation.
+- `values` (List of String) List of values in raw yaml format to pass to helm.
+- `verify` (Boolean) Verify the package before installing it.
+- `version` (String) Specify the exact chart version to install. If this is not specified, the latest version is installed.
+- `wait` (Boolean) Will wait until all resources are in a ready state before marking the release as successful.
```
Comment (merusso) [inline docs/resources/release.md:60]: I see that this new Markdown doc no longer includes the default values for variables. The defaults were documented in the prior doc versions.
